### ISSUE TITLE
Higher iterations_per_loop values in TPU training lead to NaN gradients!

### ISSUE THREAD

**Issue by mrezak** [tensorflow/tensorflow#21108]
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Cloud Platform (Linux Debian)
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
TensorFlow installed from (source or binary):Binary
TensorFlow version (use command below):1.9
Python version: 2.7
Bazel version (if compiling from source):
GCC/Compiler version (if compiling from source):
CUDA/cuDNN version: NA (TPU)
GPU model and memory: NA (TPU)
Exact command to reproduce:

### Describe the problem
I am using TPU and my model trains properly with `tpu_config=tf.contrib.tpu.TPUConfig(iterations_per_loop=20)` but when I increase `iterations_per_loop` to say 50 or 100, the gradients becomes NaN during the training (see the error below). I consistently see this behavior with higher values of `iterations_per_loop` while lower values (e.g <20) never lead to NaN gradients. 

I tried to set `num_shards=1` but the problem is the same.

 
### Source code / logs
```
InvalidArgumentError (see above for traceback): Gradient for ic_enc/ic_enc_2_ics/logvar/b:0 is NaN : Tensor had NaN values
	 [[Node: CheckNumerics_11 = CheckNumerics[T=DT_FLOAT, message="Gradient for ic_enc/ic_enc_2_ics/logvar/b:0 is NaN", _device="/job:worker/replica:0/task:0/device:CPU:0"](Read_13/ReadVariableOp)]]
```

**Comment by angersson** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new), including code that can replicate the bug. Thanks!
